### ISSUE TITLE
Containers

### ISSUE THREAD

**Issue by lars-t-hansen** [NAICNO/Jobanalyzer#300]
This came up briefly at the NRIS meeting, but further investigation would be useful: what about containers?  If an application is packaged and run as a container, will sonar need to run inside the container (and if so where do the data go?) or will sonar running outside the container be sufficiently useful to tell us something about what goes on inside (and if so what are the application name and similar attributes?).

**Comment by Sabryr** [comment]
Container users can be broadly categorized as 
1. Users created their own container image or containers created using recipes provided by other (e.g. docker file)
2. User who pull container images and did not involve in building them.
For the first case, may be it is possible to have jobanalyzer installed in the container, for the second type this is  not possible. 

To write logs the storage location needs to bind to container. Home directory is usually a good location for this exchange, 

In any case, even though in some-cases it is  possible to perform the analysis inside the container, IMO I do not think we should do that. Treating the container as a single application is adequate.  

If a container triggers alarms, then we can identify the user and decide further actions. This could be that jobanalyzer is used inside the container.

**Comment by lars-t-hansen** [comment]
The main issue for me is whether sonar running outside the container is able to say anything interesting about the memory usage inside the container.  It depends on how the container manages memory, probably.

**Comment by Sabryr** [comment]
Correct. First stage may be catch the containers that trigger alarms, next stage drill down. If we manage to get some more people involved soon, then we can increase the priority of this.
